e2e/config/vhd.go

```diff
@@ -104,12 +104,13 @@ var (
 	// if we ever want to update this then we'd need to run a new VHD build using private package overrides
 	VHDUbuntu2204Gen2ContainerdPrivateKubePkg = &Image{
 		// 2204Gen2 is a special image definition holding historical VHDs used by agentbaker e2e's.
-		Name:    "2204Gen2",
-		OS:      OSUbuntu,
-		Arch:    "amd64",
-		Version: "1.1704411049.2812",
-		Distro:  datamodel.AKSUbuntuContainerd2204Gen2,
-		Gallery: imageGalleryLinux,
+		Name:                     "2204Gen2",
+		OS:                       OSUbuntu,
+		Arch:                     "amd64",
+		Version:                  "1.1704411049.2812",
+		Distro:                   datamodel.AKSUbuntuContainerd2204Gen2,
+		Gallery:                  imageGalleryLinux,
+		UnsupportedKubeletNodeIP: true,
```

Comment:
It'd be good to have a comment as to why we need this flag.

Reply:
yeah... I guess we need it for this VHD since in the scriptless case we'd be using the original provisioning scripts baked onto it, which don't support getting kubelet node IP from IMDS (this image was built early 2024)

Reply:
Some tests checks reference to a VHD objects before performing a validation.
In my tests I built a new VHD, so the reference check doesn't work and test fails.

In general, I think it's a cleaner way to skip some default validation checks on some edge cases.